file_paths.md

```diff
@@ -0,0 +1,123 @@
+Puppet Installation Layout
+==========================
+
+This table specifies the file paths in a Puppet installation and the corresponding settings, denoted as `:setting`. The 3.x column shows the difference of the specification vs the actual 3.x implementation.
+
+puppet-agent
+------------
+
+    Path                                  Setting                        3.x
+    /etc/puppetlabs                                                      n/a
+
+    /etc/puppetlabs/code                  # :codedir                     contents moved from :confdir
+        environments                      # :environmentpath
+        hiera.yaml                        # :hiera_config
+        hieradata                         # n/a
+        modules                           # user modulepath
+
+    /etc/puppetlabs/mcollective
+        client.cfg
+        facts.yaml
+        server.cfg
+
+    /etc/puppetlabs/puppet                # :confdir                     /etc/puppet
+        auth.conf                         # :rest_authconfig
+        autosign.conf                     # :autosign
+        binder_config.yaml                # :binder_config
+        csr_attributes.yaml               # :csr_attributes
+        custom_trusted_oid_mapping.yaml   # :trusted_oid_mapping_file
+        device.conf                       # :deviceconfig
+        fileserver.conf                   # :fileserverconfig
+        puppet.conf                       # :config
+        routes.yaml                       # :route_file
+        ssl                               # :ssldir                      /etc/puppet/ssl
+
+    /opt/puppetlabs/bin                   # symlink targets of puppet related binaries, excluding ruby, openssl
```

Comment:
I might drop ", excluding ruby, openssl" since it's not an exhaustive list of what is not symlinked.